### DISCUSSION TITLE
Issue when pool is empty.

### DISCUSSION THREAD

**Discussion by ozgurerdogan** [psy0rz/zfs_autobackup#181] [General]
I am using healthchecks.io to monitor if backup fails. When there is no dataset in a pool, it creates following log:
! No source filesystems selected, please do a 'zfs set autobackup:s2d1=true' on the source datasets you want to select.
which causes healthchecks to trigger warning. 

So is it possible if pool is empty, just igrone it?

**Comment by psy0rz** [comment]
Perhaps, but maybe for now you can script it?
```
zfs list -H -o name -r $POOL |grep / >/dev/null && zfs-autobackup ...
```

This will only backup if there are datasets in that pool.

I could add a check, but its not super trivial to do it in a reliable way: Its needs to lookup all the pools, check if there are no datasets and check if the pool-dataset for each pool is really empty. And then still it could be that a sysadmin wants to begin backuping from the moment the pool was still empty. 

It seems if someone is automating it, like you, it would be better to add some scripting to do the checks that suit your environment.

**Reply by ozgurerdogan** [reply]
Ok I tried so : 

`ssh 1.2.3.4 "zfs list -H -o name -r rpool/data | grep / >/dev/null && zfs-autobackup -v  s2d1 D6 --keep-source=30 --keep-target=30 --strip-path=2 --clear-mountpoint --clear-refreservation --ignore-transfer-errors --no-holds --snapshot-format %Y-%m-%d-%H:%M:%S --force --destroy-missing=10d";
curl --retry 3 https://hc-ping.com/a0f528a0-5ba3-4ec9-929d-fea0add822f0/$?;
`
This is a proxmox server.
Still logs same text. Am I missing something?
